### DISCUSSION TITLE
How do I use scan and replace of special characters

### DISCUSSION THREAD

**Discussion by dhfreinkel** [codefori/vscode-ibmi#916] [IBM i]
How do I do a scan and replace?

Example Array[x] to be Array(x).

The system hints at needing a leading / but it responds no matches.

**Comment by worksofliam** [comment]
@dhfreinkel Do you mean a scan and replace in RPGLE? 

Like `%scanrpl` perhaps?

**Reply by chrjorgensen** [reply]
If you search using regular expression, you must specify \ before the [ and ] characters, which have special meaning in regex.

**Reply by worksofliam** [reply]
Oh. I see. @dhfreinkel means scan and replace inside of VS Code!

**Comment by dhfreinkel** [comment]
I use Ctrl F and then try to replace characters in the source code.

 

Darryl Freinkel

A4G

Telephone: 770.321.8562 Mobile: 678.355.8562

**Comment by dhfreinkel** [comment]
I tried that. It did not work.

 

Darryl Freinkel

A4G

Telephone: 770.321.8562 Mobile: 678.355.8562

**Comment by andreas-prouza** [comment]
Are you doing this in that way?:
![image](https://user-images.githubusercontent.com/65070910/198196972-db251cff-924e-4a3c-985e-5747459b17e6.png)

**Comment by dhfreinkel** [comment]
Yes, I used array[x]. I changed it to array(x).

 

Thanks

 

Darryl Freinkel

A4G

Telephone: 770.321.8562 Mobile: 678.355.8562
